### PR TITLE
Fixed #700: redesign meal defaults/schedule UI

### LINKED ISSUE #700: Meal defaults and Meal schedule should be separate elements in the UI
### Expected Behaviour
- Meal defaults and meal schedule are two different concepts and the UI should reflect this.
- Irrespective of delivery type (ongoing and episodic) the client preferences should always show the meal defaults, that is **what** the client said he wants in **his** meal for specific weekdays.
- For an ongoing delivery type only, the client preferences should also show the meal schedule, that is **when** (= on which weekdays) he wants **his** meal delivered.

(This is an addition to issue #688, based on requirements given by Jean-François on 2017-03-17)
### Actual Behaviour
- The UI only offers Meal schedule which is confusing.

### LINKED ISSUE #710: Behaviour inconsistent in client selection on the New Orders page
### Expected Behaviour

Selecting a client on the new orders page should either:

 * complete an AJAX GET request to obtain necessary information from the API, or
 * complete a POST + full page reload to obtain necessary information, as if selecting a new delivery date,

but not both.

### Actual Behaviour

On client selection, a GET request is fired [here](https://github.com/savoirfairelinux/sous-chef/blob/dev/src/frontend/js/order.js#L82) which tries to obtain information about meal defaults but in reality appears to do nothing. This AJAX call enters into conflict with the one [here](https://github.com/savoirfairelinux/sous-chef/blob/dev/src/frontend/js/multidatespicker.js#L64), which is never actually executed due to a bug.

We should decide which of these calls to keep and fix, and which one to scrap.

### Steps to reproduce

* Open developer tools
* Click Orders > New Order
* Select a client
* Observe the GET request being fired, but the server returns an XHTML document instead of the expected JSON, etc etc

### PR DESCRIPTION
**Pull request by lingxiaoyang**
### Changes proposed in this pull request:

**Note: This PR contains multiple cleanups related with meal defaults/schedule functionality.**

1. Redefine the returning values of `client.meals_default` and `client.meals_schedule` properties.
    1. The validation logic of `client.meals_default` is moved to form `member.forms.ClientRestrictionsInformation`, and `client.meals_default` returns the database value in an "as is" manner: `None` is `None` and zero is zero, which permits the distinction of these two values as "None -- the client said nothing" and "0 -- the client said he didn't want".
    2. `client.meals_schedule` is simply changed to a filter of `client.meals_default` based on delivery days.
    3. Update tests.
    4. Fix `order.models.OrderManager.auto_create_orders()` regarding this change.
2. New tabular UI for editing `client.meals_default`:
    1. Viewing (links to [screenshot1](https://cloud.githubusercontent.com/assets/8630726/24672599/55ef794c-1943-11e7-9493-283e0dcb6b1f.png), [screenshot2](https://cloud.githubusercontent.com/assets/8630726/24672624/65ac4086-1943-11e7-9d45-5a04468726aa.png), [screenshot3](https://cloud.githubusercontent.com/assets/8630726/24672643/74af0c30-1943-11e7-9c37-1f131a9941b2.png)): delivery days are displayed in the same line of "Ongoing". The table shows the quantities of different components (for main dish, also the abbreviated size for a quick view). In the table, the truck icon is displayed for delivery days for ongoing clients. Tooltips are shown when the user hovers the truck icon, size L/R, or not-provided field.
    2. Editing ([link to screenshot](https://cloud.githubusercontent.com/assets/8630726/24672682/98d40f3e-1943-11e7-8748-37a3b4ab960e.png)): meal defaults are edited in a scrollable table. Null values (blank values) are permitted, if the client is episodic or the ongoing client doesn't want delivery on a particular day (otherwise, [a validation error](https://cloud.githubusercontent.com/assets/8630726/24670283/6e9c64a8-193b-11e7-9314-56abef602a92.png)). If main dish quantity is provided, the size is required (otherwise, [a validation error](https://cloud.githubusercontent.com/assets/8630726/24670324/95a5001e-193b-11e7-8287-8e1040554a96.png)).
3. Refactoring template tags:
    1. `member_extras` --> `sous_chef_extras` because the tags are shared by all modules.
    2. Remove `{% cell %}` and `{% size %}`. They are confusing and replaced with another implementation.
4. Delete unused template files.
5. Delete unused client methods: 
    1. `client.get_meal_defaults`: we use `client.meals_default` instead.
    2. `client.set_meal_defaults`: we are converting the data format directly in view functions.
    3. `client.get_meal_prefs` and `client.set_meal_prefs`: the old way to store client meal preferences (regardless of weekdays). Not used by the `datamigration` module.
    4. Remove the few occurrences of these methods. In the meantime, fix #710 as we don't use these code anymore.
6. Update Semantic UI and jQuery versions to the latest minor releases.

### Status

- [x] READY
- [ ] HOLD
- [ ] WIP (Work-In-Progress)

### How to verifythis change

http://localhost:8000/member/view/100/preferences

### Deployment notes and migration

none

### New translatable strings

 - [x] Included

### Additional notes

none

### CODE REVIEW

**Comment by lingxiaoyang** [comment]
@JeanFrancoisFournier  Please see if this meets your requirement. Let me know your comments!

File: src/order/models.py [hunk 1]
```diff
@@ -34,7 +34,7 @@
 ORDER_STATUS_ORDERED = ORDER_STATUS[0][0]
 
 SIZE_CHOICES = (
-    ('', _('Serving size')),
+    ('', ''),
```
Comment (erozqba) [inline src/order/models.py]: Hi @lingxiaoyang, I think this change implies a migration that you don't have added in your PR. Could you check this please?

**Comment by lingxiaoyang** [comment]
@erozqba  Thank you for pointing out this!

I added the migration and rebased this PR.

**Comment by erozqba** [comment]
@aaboffill are you Ok with all the changes or you have some suggestions? If not, I'm ok accepting this PR.

**Pull request merged**